src/common/verbose.hpp

```diff
@@ -63,6 +63,21 @@ struct const_expr_value_t {
 
 } // namespace utility
 
+#if defined(__GNUC__) || defined(__clang__)
+#define VERBOSE_FMT_ATTR __attribute__((format(printf, 1, 2)))
+#else
+#define VERBOSE_FMT_ATTR
+#endif
+
+// This is a safeguard for verbsoe macros against mismatches between
+// format specifiers and passed arguments that could lead to runitme failures.
+// Any discrepancies will emit warnings during build time for GCC,
+// clang builds.
+static inline int _format_type_checker(const char *, ...) VERBOSE_FMT_ATTR;
+static inline int _format_type_checker(const char *, ...) {
+    return 0;
+}
```

Comment:
Trailing underscores are not welcome by clang-tidy. There're some spelling typos that I fixed in the comment as well.
```suggestion
// This is a safeguard for verbose macros against mismatches between
// format specifiers and passed arguments that could lead to runtime failures.
// Any discrepancies will emit warnings during build time for GCC and
// clang builds.
#if defined(__GNUC__) || defined(__clang__)
static inline int format_type_checker(const char *, ...) __attribute__((format(printf, 1, 2)));
#else
static inline int format_type_checker(const char *, ...) {
    return 0;
}
#endif
```

Reply:
Updated. Thanks.